pyccel/codegen/printing/ccode.py

```diff
@@ -252,7 +253,12 @@ def __init__(self, parser, prefix_module = None):
 
     def get_additional_imports(self):
         """return the additional imports collected in printing stage"""
-        return self._additional_imports
+        return [i.source if isinstance(i, Import) else i for i in self._additional_imports]
+
+    def get_additional_import_objects(self):
+        """return the additional imports collected in printing stage"""
+        return [i if isinstance(i, Import) else Import(i, Module(i, (),())) \
+                for i in self._additional_imports]
```

Comment:
Does `_additional_imports` contain both `str` and `Module` objects now?

Reply:
In `CCode` yes, this seemed easier for the developers than needing to remember to create an empty `Module` everywhere we want to import a C header. It could probably be improved by declaring all the modules we will use as global variables

Reply:
Do you want me to do this?

Reply:
Sorry, I cannot understand your explanation. Why are you talking about headers? And what about global variables?

This is my simple observation: after introducing a new `Module` class, I would expect an homogeneous list of `str` to become an homogeneous list of `Module`. Mixing `str` and `Module` objects seems a bit messy to me.

Reply:
Up till now all the imports in `_additional_imports` are imports which import a whole `Module` (or a C header file) i.e. `math`, `stdlib`, etc

I think the developers won't want to write functions like:
```
    def _print_MathIsnan(self, expr):
        """Convert a Python expression with a math isnan function call to C
        function call"""
        # add necessary include
        self._additional_imports.add(Import('math', Module('math', (),())))
        arg = expr.args[0]
        if arg.dtype is NativeInteger():
            code_arg = self._print(NumpyFloat(arg))
        else:
            code_arg = self._print(arg)
        return "isnan({})".format(code_arg)
```
I think this will be forgotten all the time.

My suggestion is to create `Import` objects for all the imports that are done in this file and to have them as global variables (as opposed to `_additional_imports` which is a class variable) then we can do:
```
self._additional_imports.add(math_import)
```

Reply:
But it means I have to find them all

Reply:
Fixed 6373f476

Reply:
Thanks for the explanation! It is a good idea and it looks clean.